File: app/src/main/java/com/akhbulatov/githubsample/ui/main/userdetailsroot/userdetails/UserDetailsView.java
package com.akhbulatov.githubsample.ui.main.userdetailsroot.userdetails;

import android.support.annotation.NonNull;

import com.akhbulatov.githubsample.models.UserDetails;
import com.akhbulatov.githubsample.ui.global.BaseView;

interface UserDetailsView extends BaseView {
    void showContentLayout(boolean show);

    void showUserDetails(@NonNull UserDetails userDetails);

    void showLoadingProgress(boolean show);

    void showToFavoritesAdded();

    void backToUser();
}

File: app/src/main/java/com/akhbulatov/githubsample/ui/main/userdetailsroot/userdetails/UserDetailsView.kt
package com.akhbulatov.githubsample.ui.main.userdetailsroot.userdetails

import com.akhbulatov.githubsample.models.UserDetails
import com.akhbulatov.githubsample.ui.global.BaseView

interface UserDetailsView : BaseView {
    fun showContentLayout(show: Boolean)
    fun showUserDetails(userDetails: UserDetails)
    fun showLoadingProgress(show: Boolean)
    fun showError(error: String)
    fun showToFavoritesAdded()
    fun backToUser()
}